packages/client/src/MinimalClientContext.ts

```diff
@@ -24,13 +26,18 @@ import type { OntologyProvider } from "./ontology/OntologyProvider.js";
 export interface MinimalClient extends ClientContext<MinimalClientParams> {}
 
 export type MinimalClientParams = {
-  metadata: {
-    userAgent: string;
-    ontologyApiName: string;
-
-    /** @deprecated this is only for transitional code. Do not rely on this for production */
-    ontologyRid: string;
-  };
+  metadata: MinimalClientMetadata<any>;
 
   provider: OntologyProvider;
 };
+
+export interface MinimalClientMetadata<V extends VersionString<any, any, any>>
+  extends OntologyMetadata<V>
+{
+  expectsClientVersion: V;
```

Comment:
MinimalClientMetadata requires this field. It is not required in the api to support v1 still working, but we can rely on v2 having this set.